src/xref2/ref_tools.ml

```diff
@@ -388,6 +390,33 @@ module EC = struct
     Ok (`Extension (parent', name))
 end
 
+module ED = struct
+  (** Extension decl *)
+
+  let in_env env name =
+    env_lookup_by_name Env.s_extension name env
+    >>= fun (`Extension (id, _, te)) ->
+    match te.constructors with
+    | [] -> assert false
+    | c :: _ ->
```

Comment:
I think this pick the first constructor as the name of the extension in the case that you add multiple in one go. I am not sure; this probably needs a comment.

Reply:
I think this : "this is how extension declarations are given an ID, extension declaration always have a first constructor" would stand to be explained here.

Reply:
I fixed the conflicts and added a comment.

Reply:
I would say that this comment would make sense where the type is defined (in `path_types.ml`), and/or where the function to create those ids is defined (in `paths.ml`, value `Identifier.Mk.extension_decl`).

Reply:
(and, why not also in the `ED.in_env` function, since this is where Emile has commented about the lack of comment?)

Reply:
I added more comments.